app/src/main/java/ch/epfl/sdp/peakar/user/profile/NewProfileActivity.java

```diff
@@ -131,4 +192,119 @@ private void expandSelected(boolean expand) {
             selectedCollected.findViewById(R.id.collected_date).setVisibility(expand ? View.VISIBLE:View.GONE);
         }
     }
-}
\ No newline at end of file
+
+    /**
+     * On sign out button click
+     */
+    public void signOutButton(View view) {
+        AuthService.getInstance().signOut(this);
+        finish();
+    }
+
+    /**
+     * On change username button click
+     */
+    public void changeUsernameButton(View view) {
+        // Hide username
+        findViewById(R.id.profile_username).setVisibility(View.GONE);
+
+        // Hide change button
+        findViewById(R.id.profile_change_username).setVisibility(View.GONE);
+
+        // Show edit text
+        EditText changeUsername = findViewById(R.id.profile_username_edit);
+        changeUsername.setVisibility(View.VISIBLE);
+        changeUsername.requestFocus();
+
+        InputMethodManager imm = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
+        imm.showSoftInput(changeUsername, InputMethodManager.SHOW_IMPLICIT);
+
+        changeUsername.setOnKeyListener((v, keyCode, event) -> {
+            if ((event.getAction() == KeyEvent.ACTION_DOWN) &&
+                    (keyCode == KeyEvent.KEYCODE_ENTER)) {
+                confirmUsernameButton();
+                return true;
+            }
+            return false;
```

Comment:
Return true for key_up and down button press.
```suggestion
            }
            return true;
```